Main.elm

```diff
@@ -0,0 +1,256 @@
+            Nothing
+
+
+init : Flags -> Navigation.Location -> ( Model, Cmd Msg )
+init flags location =
+    let
+        authCode =
+            extractAuthCode location
+    in
+        { server = "https://mamot.fr"
+        , registration = flags.registration
+        , client = flags.client
+        , userTimeline = []
+        , publicTimeline = []
+        , errors = []
+        , location = location
+        }
+            ! [ initCommands flags.registration flags.client authCode ]
+
+
+initCommands : Maybe Mastodon.AppRegistration -> Maybe Mastodon.Client -> Maybe String -> Cmd Msg
+initCommands registration client authCode =
+    Cmd.batch <|
+        case authCode of
+            Just authCode ->
+                case registration of
+                    Just registration ->
+                        [ Mastodon.getAccessToken registration authCode |> Mastodon.send AccessToken ]
+
+                    Nothing ->
+                        []
+
+            Nothing ->
+                case client of
+                    Just client ->
+                        [ loadTimelines client ]
+
+                    Nothing ->
+                        []
+
+
+registerApp : Model -> Cmd Msg
+registerApp { server, location } =
+    let
+        appUrl =
+            location.origin ++ location.pathname
+    in
+        Mastodon.register
+            server
+            "tooty"
+            appUrl
+            "read write follow"
+            |> Mastodon.send AppRegistered
+
+
+saveClient : Mastodon.Client -> Cmd Msg
+saveClient client =
+    Mastodon.clientEncoder client
+        |> Encode.encode 0
+        |> Ports.saveClient
+
+
+saveRegistration : Mastodon.AppRegistration -> Cmd Msg
+saveRegistration registration =
+    Mastodon.registrationEncoder registration
+        |> Encode.encode 0
+        |> Ports.saveRegistration
+
+
+loadTimelines : Mastodon.Client -> Cmd Msg
+loadTimelines client =
+    Cmd.batch
+        [ Mastodon.fetchUserTimeline client |> Mastodon.send UserTimeline
+        , Mastodon.fetchPublicTimeline client |> Mastodon.send PublicTimeline
+        ]
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        NoOp ->
+            model ! []
+
+        ServerChange server ->
+            { model | server = server } ! []
+
+        UrlChange location ->
+            model ! []
+
+        Register ->
+            model ! [ registerApp model ]
+
+        AppRegistered result ->
+            case result of
+                Ok registration ->
+                    { model | registration = Just registration }
+                        ! [ saveRegistration registration
+                          , Navigation.load <| Mastodon.getAuthorizationUrl registration
+                          ]
+
+                Err error ->
+                    { model | errors = (toString error) :: model.errors } ! []
+
+        AccessToken result ->
+            case result of
+                Ok { server, access_token } ->
```

Comment:
Doing too much Python? ;-) I suppose that `access_token` should be `accessToken` here?